src/modules/site-v2/templates/_scripts/utils.js

```diff
@@ -98,3 +98,54 @@ function force_download(e) {
       a.click();
     })
 }
+
+
+/* Create a DOM node from a string of HTML, without adding the element to the DOM yet */
+function create_node(html) {
+  const placeholder = document.createElement("div");
+  placeholder.innerHTML = html;
+  return placeholder.firstElementChild;
+}
+
+
+/* Flash a message to the user without reloading the page.
+ * This function is roundabout to prevent code injection.
+ */
+function flash_message(message, full_msg_link=null, full_msg_body=null) {
+
+  // Define the template for an alert popup using a static string
+  {%- with msg='', category='danger' %}
+  const raw_html = `{% include '_includes/alert.html' %}`;
+  {%- endwith %}
+
+  // Create as a new DOM node, and insert the desired message as text
+  // Inserting as text protects against code injection
+  const node = create_node(raw_html);
+  node.firstElementChild.innerText = message;
+
+  // If both full message fields are provided, add as a link & collapse dropdown
+  if (full_msg_link && full_msg_body) {
+    node.firstElementChild.innerText += ' ';
+
+    // Create a unique ID to link the trigger & collapse elements
+    const id = `err-dropdown-${(new Date()).getTime()}`;
```

Comment:
prefer `error` vs `err`

Reply:
Fixed in [b137d57](https://github.com/AndersenLab/CAENDR/pull/402/commits/b137d5709aff41d74bc8177241ca161ae2e3d02d)